core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -161,15 +167,20 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
                 }
             }
             this.translog = translog;
-            manager = createSearcherManager();
-            this.searcherManager = manager;
-            this.versionMap.setManager(searcherManager);
+            searcherManager = createSearcherManager();
+            this.searcherManager = searcherManager;
+
+            internalSearcherManager = createSearcherManager();
```

Comment:
I think the internalSearcherManager doesn't need the same searcher factory as we use for the external one since it doesn't need to do warming etc? I think it should be very lean?